lib/senkyoshi/models/file_resource.rb

```diff
@@ -0,0 +1,23 @@
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  ##
+  # Class to represent a resource constructed from a single 'dat' file.
+  ##
+  class FileResource < Resource
+    attr_reader(:id)
+
+    def initialize(id = nil)
+      @id = id
+    end
+
+    def self.from(xml = nil, pre_data = nil, _resource_xids = nil)
+      resource = new(pre_data[:file_name])
+      resource.iterate_xml(xml, pre_data)
```

Comment:
Perhaps I'm not reading the code correctly. It looks to me that we're creating a new instance of `FileResource` on line 15 and then calling `iterate_xml` on that instance. 
`FileResource`'s implementation of `iterate_xml` is 
```ruby
def iterate_xml(_xml, _pre_data)
  self
end
```
Neither of the arguments passed to it are used. Is that correct?